backend/app/models/user.rb

```diff
@@ -13,7 +13,7 @@ class User < ApplicationRecord
   has_many :investors, through: :favourite_investors
   has_many :favourite_open_calls, dependent: :destroy
   has_many :open_calls, through: :favourite_open_calls
-  has_one :owner_account, class_name: "Account", foreign_key: "owner_id", dependent: :restrict_with_error
+  has_one :owner_account, class_name: "Account", foreign_key: "owner_id"
```

Comment:
I had to remove `dependent: :restrict_with_error`, because it prevented proper account deletion, but Backoffice disables button for user deletion if user is account owner and just to be save, I have put also
```
return redirect_to(backoffice_users_path, status: :see_other, alert: I18n.t("errors.messages.user.is_owner")) if @user.owner_account.present?
```
to User controller.

Reply:
Are we safe on the API side of user removal?

Reply:
I understand we restrict removing user who is account owner, right?

Reply:
Yes, we are safe on API side, it is handled by cancancan and there is even test for this use case